src/libs/fullstory/index.ts

```diff
@@ -46,40 +53,38 @@ const FS = {
     consent: (c: boolean) => FullStory('setIdentity', {consent: c}),
 
     /**
-     * Initializes the FullStory session with the provided session information.
+     * Initializes the FullStory metadata with the provided metadata information.
      */
-    consentAndIdentify: (value: OnyxEntry<Session>) => {
+    consentAndIdentify: (value: OnyxEntry<UserMetadata>) => {
         try {
-            FS.onReady().then(() => {
-                const session: UserSession = {
-                    email: value?.email,
-                    accountID: value?.accountID,
-                };
-                // set consent
-                FS.consent(true);
-                FS.fsIdentify(session);
+            Environment.getEnvironment().then((envName: string) => {
+                if (CONST.ENVIRONMENT.PRODUCTION !== envName) {
+                    return;
+                }
+                FS.onReady().then(() => {
+                    FS.consent(true);
+                    FS.fsIdentify(value);
+                });
             });
         } catch (e) {
             // error handler
         }
     },
 
     /**
-     * Sets the FullStory user identity based on the provided session information.
-     * If the session does not contain an email, the user identity is anonymized.
-     * If the session contains an email, the user identity is defined with the email and account ID.
+     * Sets the FullStory user identity based on the provided metadata information.
+     * If the metadata does not contain an email, the user identity is anonymized.
+     * If the metadata contains an accountID, the user identity is defined with it.
      */
-    fsIdentify: (session: UserSession) => {
-        if (typeof session.email === 'undefined') {
-            // anonymize FullStory user identity session
+    fsIdentify: (metadata: UserMetadata | null) => {
+        if (!metadata?.accountID) {
```

Comment:
Same question here, wondering what would happen if metadata is null and we then access `metadata.accountID` in else

Reply:
same answer as the other comment